### PR TITLE
Assignment 3: Add assignment page + button on gradebook homepage

### PR DESCRIPTION
**Pull request by andrewpaulino**
# Description

We want to add the ability to add a new assignment to our gradebook. To do this we did two things

1. Added a button on listing page
2. Added a component to host our assignment

This way we can save our assignment and also populate a message.

Some other features

- Validation of form, the ability to check if form is complete
- MUI components for form

# Screenshots

Demo (I deliberetly put the due date in the past to showcase it in the needs grading section):

https://user-images.githubusercontent.com/29711542/226808131-0e0e8389-a82a-498a-bf3f-8aa249b189c3.mp4

image

image

### CODE REVIEW

File: src/components/AddAssignment.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                        Create Assignment
+                    </Button>
+                </div>   
+                <ToastContainer autoClose={1500} /> 
+            </form>
+        </div>
+    )
+}
+
+export default AddAssignment;
```
Comment (lyratadev) [inline src/components/AddAssignment.js]: Make sure to add a new line here.

File: src/App.js [hunk 2]
```diff
@@ -20,6 +21,7 @@ function App() {
        <Switch>
         <Route exact path='/' component={Assignment} />
         <Route path='/gradebook' component={Gradebook} />
+        <Route path='/add/assignment' component={AddAssignment} />
```
Comment (lyratadev) [inline src/App.js:24]: Should the front-end routes be the same as the backend api's?

File: src/components/AddAssignment.js (hunk 1 above)
Comment (andrewpaulino) [reply src/components/AddAssignment.js]: Will do

File: src/App.js (hunk 2 above)
Comment (andrewpaulino) [reply src/App.js:24]: Yes, it's fine for that because it is not hitting the same resource.

**Review by lyratadev — APPROVED** [review]

**Pull request merged**